answer_rocket/chat.py

```diff
@@ -0,0 +1,187 @@
+import os
+import logging
+from typing import Literal, Callable, Optional
+
+import openai
+from sgqlc.types import Variable, non_null, String, Arg
+from sgqlc.operation import Fragment
+
+from answer_rocket.auth import AuthHelper
+from answer_rocket.graphql.client import GraphQlClient
+from answer_rocket.graphql.schema import (LLMApiConfig, AzureOpenaiCompletionLLMApiConfig,
+                                          AzureOpenaiEmbeddingLLMApiConfig, OpenaiCompletionLLMApiConfig,
+                                          OpenaiEmbeddingLLMApiConfig)
+
+logger = logging.getLogger(__name__)
+
+ModelType = Literal['GPT3_5', 'GPT4', 'Embedding']
+ApiType = Literal['AZURE', 'OPENAI']
+
+LLMConfigType = LLMApiConfig | AzureOpenaiEmbeddingLLMApiConfig | AzureOpenaiCompletionLLMApiConfig | \
+                OpenaiCompletionLLMApiConfig | OpenaiEmbeddingLLMApiConfig
+
+
+class Chat:
```

Comment:
nit: at some point I think the class itself might benefit from a docstring, but probably not a big deal for now